src/navigation/RootNavigator.tsx

```diff
@@ -0,0 +1,50 @@
+import { NavigationContainer } from '@react-navigation/native';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import React from 'react';
+import GalleryScreen from '../screens/GalleryScreen';
+import ShareScreen from '../screens/ShareScreen';
+import BottomTabIcon from './BottomTabIcon';
+
+const BottomTab = createBottomTabNavigator();
+
+export default function RootNavigator() {
+  return (
+    <NavigationContainer>
+      <BottomTab.Navigator
+        tabBarOptions={{
+          showLabel: false,
+          style: {
+            backgroundColor: '#fff',
+            height: 70,
+            borderTopColor: '#555',
+          },
+        }}
+      >
+        <BottomTab.Screen
+          name="Gallery"
+          component={GalleryScreen}
+          options={{
+            tabBarIcon: ({ focused }) =>
+              BottomTabIcon({
+                title: 'Gallery',
+                iconSource: require('../../assets/gallery-tab-icon.png'),
+                isFocused: focused,
+              }),
+          }}
+        />
+        <BottomTab.Screen
+          name="Share"
```

Comment:
I'd rather prefer `StackNavigator` for these two screens. By default, the gallery screen is open. When clicked on  a single photo, a "Image Details" screen is opened and stack navigator automatically gives you the "Go Back" navigation button. This is more natural behavior than jumping to second tab when clicked something on the first tab
 If we want to have tabs too (they're looking cool), the second tab should be something like settings or I don't know what. For now it may be "Playground"/"Debug Screen" used for testing purposes.

Technically, we can do it in two (or three) ways:
- nest `<StackNavigator>` inside `<BottomTabNavigator>` - in this case tabs are always visible, and the "Go Back" button is visible only when in first tab ("Gallery").
- nest `<BottomTabNavigator>` inside `<StackNavigator>` - in this case tabs are only visible on the "Gallery Screen" and they disappear when you enter another screen inside stack.
- multiple nesting / other combinations - gives more flexibility but overcomplicated for now

Some time ago I made a complex example of multiple navigators nested:
https://github.com/barthap/MemeBrowser/blob/08d56414b851e9e2a21c8cff3d1ecf44a303d000/src/AppMain.tsx#L39
When you click e.g. on `DrawerNavigator` or `RootNavigator`, it redirects you to its definition.

Reply:
I went with the first option you proposed (`<StackNavigator>` is nested inside `<BottomTabNavigator>`).